lib/src/rules/use_late.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/analysis/features.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+import '../util/dart_type_utilities.dart';
+
+const _desc = r'Use late for private members with non-nullable type.';
+
+const _details = r'''
+
+Use late for private members with non-nullable type to avoid null checks.
```

Comment:
Are we attempting to avoid null checks in order to improve performance or for readability?

Reply:
Mainly readability because I don't have number about perf (as there are less null checks I guess it will be better).

Reply:
Based on Erik's comment, it sounds like code should have about the same performance either way. It would be good to clarify in the details that the purpose is to make it clear in the code that a field / variable is not expected to be `null`. Users who are new to null safety might not yet understand the value of `late`.

Reply:
Comment updated

Reply:
Readability.

It's unlikely to be measurably more performant in general. 
In some cases we might be able to eliminate the initialization checks of a late variable (say when it's initialized in the body of the only generative constructor), but then, we could probably eliminate null checks in the same cases if we wanted to.